### ISSUE TITLE
Magic string for tar.gz archive not found in RPMs

### ISSUE THREAD

**Issue by AdrianWennberg** [goreleaser/nfpm#62]
I am creating RPMs and trying to inspect them using the python [rpmfile](https://github.com/srossross/rpmfile) library, but I am getting the following failure:
```
Traceback (most recent call last):
  File "rpm_unit_test.py", line 509, in test_rpmfile
    print r.getmembers()
  File "/usr/lib/python2.7/site-packages/rpmfile/__init__.py", line 112, in getmembers
    magic = g.read(2)
  File "/usr/lib64/python2.7/gzip.py", line 267, in read
    self._read(readsize)
  File "/usr/lib64/python2.7/gzip.py", line 302, in _read
    self._read_gzip_header()
  File "/usr/lib64/python2.7/gzip.py", line 196, in _read_gzip_header
    raise IOError, 'Not a gzipped file'
IOError: Not a gzipped file
```

The reason for the failure is that rmfile cannot find the "Magic string" to identify the gzip file within the RPM:
```
    def _read_gzip_header(self):
        magic = self.fileobj.read(2)
        if magic != '\037\213':
            raise IOError, 'Not a gzipped file'
        ...
```
This "magic string" should show up as a part of the RPM file as the last part of the file should be a gzipped archive that holds the actual files of the package. [RPM spesification](http://ftp.rpm.org/max-rpm/s1-rpm-file-format-rpm-file-format.html#S2-RPM-FILE-FORMAT-THE-ARCHIVE)

Comparing the rpm to one built using FPM using a hex editor shows that the magic string is indeed missing. I'm not sure how or why this is missing, but it indicates that something is weird with the way NFPM builds RPMs. I'll keep looking into this and update here with my findings.

**Comment by AdrianWennberg** [comment]
It seems like it's not just the magic string that is missing, but the entire gzip file is incorrectly compressed. The RPM can still be unpacked using a tool like rpm2cpio so it seems like it is still valid, even though it doesn't conform to the specifications.

Also, if someone could verify this behavior, that would be great. 

Steps to reproduce:
1. Build an RPM using nfpm
2. run `xxd -p filename.rpm | tr -d '\n' | grep -cm 1 '1f8b'`

This will output a 0 of no magic string was found, and a 1 if one was found.

**Comment by AdrianWennberg** [comment]
Oh, I just got what the error seems to be. RPMs support different compression methods and the default (for Fedora) seems to be `xz` [source](https://en.wikipedia.org/wiki/RPM_Package_Manager#Binary_format). I can't find any good sources on this though. The official [File format documentatiopn](https://rpm.org/devel_doc/file_format.html) is lacking, and other tools like FPM uses gzip it seems.

**Comment by caarlos0** [comment]
hmmm 

do you know if other platforms accept xz as well? Maybe we could use it instead 🤔

**Comment by AdrianWennberg** [comment]
I can't find any good documentation for this, but as far as I can tell, most platforms and tools are gonna support both types of compression. Also of note is that Fedora is [moving to zstd compression](https://fedoraproject.org/wiki/Changes/Switch_RPMs_to_zstd_compression) soon. I don't think compatibility in general will be an issue as long as NFPM supports either xz or gzip.

I think we should consider allowing the user to choose which compression method to use as an optional configuration, but by default use the system default. Issue #40 discusses manually building RPMs, which would be a bit more complex if we supported multiple compression methods, but I think it's doable.

**Comment by AdrianWennberg** [comment]
Fixed in PR #63

**Comment by caarlos0** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
